### PR TITLE
PYTHON-1371 - The tailable cursor cannot get document through __getitem__(index) on MongoDB v3.4

### PR DESCRIPTION
**Pull request by caosiyang**
Fix issue and add test case.

### CODE REVIEW

File: pymongo/cursor.py [hunk 1]
```diff
@@ -590,6 +590,7 @@ def __getitem__(self, index):
             clone = self.clone()
             clone.skip(index + self.__skip)
             clone.limit(-1)  # use a hard limit
+            clone.__query_flags &= ~CursorType.TAILABLE  # PYTHON-1371
```
Comment (behackett) [inline pymongo/cursor.py]: I haven't tested it, but I assume awaitData has the same problem. This should probably use CursorType.TAILABLE_AWAIT.

File: test/test_cursor.py [hunk 2]
```diff
@@ -1133,6 +1133,12 @@ def test_tailable(self):
 
         self.assertEqual(3, db.test.count())
 
+        # __getitem__(index)
+        cursor2 = db.test.find(cursor_type=CursorType.TAILABLE)
```
Comment (behackett) [inline test/test_cursor.py]: Test both TAILABLE and TAILABLE_AWAIT.

File: test/test_cursor.py [hunk 3]
```diff
@@ -1133,6 +1133,12 @@ def test_tailable(self):
...
+        self.assertEqual(4, cursor2[0]["x"])
+        self.assertEqual(5, cursor2[1]["x"])
+        self.assertEqual(6, cursor2[2]["x"])
+
```
Comment (ShaneHarvey) [inline test/test_cursor.py]: Can test this with `CursorType.TAILABLE_AWAIT` too? And another for `__getitem__(slice)`, eg
```python
cursor = db.test.find(cursor_type=CursorType.TAILABLE_AWAIT)
self.assertEqual([4,5], [doc["x"] for doc in cursor[0:2]])
```

File: pymongo/cursor.py (hunk 1 above)
Comment (caosiyang) [reply pymongo/cursor.py]: @behackett You are right. I miss the 'awaitData' bit.

File: test/test_cursor.py (hunk 2 above)
Comment (caosiyang) [reply test/test_cursor.py]: @behackett OK.  I will test them later.

File: test/test_cursor.py (hunk 3 above)
Comment (caosiyang) [reply test/test_cursor.py]: @ShaneHarvey Thanks for your advice. I will expand the tests later.

**Comment by caosiyang** [comment]
I commit some new code, please review again.

**Comment by ShaneHarvey** [comment]
Thanks, @caosiyang. I'm running a job on our internal CI to test against all MongoDB versions since our Travis only tests against the default installed MongoDB. I'll let you know the results.

**Comment by ShaneHarvey** [comment]
Thanks @caosiyang LGTM! Merged: https://github.com/mongodb/mongo-python-driver/commit/db57f671dcf018dc90f4b7fc8e3db75d62ee1766

**Pull request closed without merging**